### PR TITLE
add driver for Lakeshore Model 335

### PR DESCRIPTION
**Pull request by jbeckerwsi**
- Add driver for Lakeshore Model 335
- Increase range on setpoint property of Lakeshore Model 335,336,372 to support degree Celsius

### CODE REVIEW

**Comment by jbeckerwsi** [comment]
Pending test on hardware

File: qcodes/instrument_drivers/Lakeshore/Lakeshore_model_335.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+from typing import Dict
+
+import pyvisa as visa
+
+from .lakeshore_base import LakeshoreBase
+from .Lakeshore_model_336 import (
+    LakeshoreModel336Channel,
+    LakeshoreModel336CurrentSource,
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/Lakeshore/Lakeshore_model_335.py:8]: We have been slowly converting to exposing the drivers as a public api at a level above this. So could you add the driver to the `__init__` file in this folder (following the pattern of the rest) and make this file private (start the name with an underscore_

File: qcodes/instrument_drivers/Lakeshore/Lakeshore_model_335.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
+from typing import Dict
+
+import pyvisa as visa
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/Lakeshore/Lakeshore_model_335.py:3]: To not depend on implementation details of imports in the pyvisa module you should also explicitly import pyvisa.constants

I would prefer to get rid of the pyvisa as visa alias and just import pyvisa as is

File: qcodes/instrument/sims/lakeshore_model335.yaml [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+devices:
+  device 1:
+    eom:
+      GPIB INSTR:
+        q: "\r\n"
+        r: "\r\n"
+    error: ERROR
+    dialogues:
+      - q: "*IDN?"
+        r: "QCoDeS, m0d3l, 336, 0.0.01"
```
Comment (jenshnielsen) [inline qcodes/instrument/sims/lakeshore_model335.yaml:11]: Does the 335 really return this on idn?
```suggestion
        r: "QCoDeS, m0d3l, 336, 0.0.01"
```

**Review by jenshnielsen — APPROVED** [review]

**Pull request merged**